alf/bin/train_test.py

```diff
@@ -0,0 +1,131 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tempfile
+import os
+import io
+import subprocess
+from pathlib import Path
+import unittest
+import numpy as np
+
+from absl import logging
+from absl.testing import parameterized
+
+from tensorboard.backend.event_processing import event_file_loader
+from tensorboard.util import tensor_util
+
+
+def run_and_stream(cmd, cwd):
+    """Run cmd in a new process and stream its stdout and stderr output
+
+    Args:
+        cmd (list[str]): command and args to run
+        cwd (str): working directory for the process
+    """
+    logging.info("Running %s", " ".join(cmd))
+    process = subprocess.Popen(
+        cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, cwd=cwd)
+
+    while process.poll() is None:
+        with io.TextIOWrapper(process.stdout, encoding="utf-8") as text_io:
+            for line in text_io:
+                logging.info(line.strip())
+
+    if process.returncode != 0:
+        logging.error("cmd: {0} exited with code {1}".format(
+            " ".join(cmd), process.returncode))
+
+
+def get_metrics_from_eval_tfevents(eval_dir):
+    """Get metrics from tfevents in eval dir
+
+    Args:
+        eval_dir (str):
+
+    Returns:
+        list[float], list[int]: average returns, and average episode lengths
+    """
+    event_file = ''
+    for root, dirs, files in os.walk(eval_dir):
+        for file_name in files:
+            if "events" in file_name and 'profile' not in file_name:
+                event_file = os.path.join(root, file_name)
+                break
+
+    logging.info("Parse event file:%s", event_file)
+    episode_returns = []
+    episode_lengths = []
+    for event_str in event_file_loader.EventFileLoader(event_file).Load():
+        if event_str.summary.value:
+            for item in event_str.summary.value:
+                if item.tag == 'Metrics/AverageReturn':
+                    tensor = tensor_util.make_ndarray(item.tensor)
+                    episode_returns.append(float(tensor))
+                elif item.tag == 'Metrics/AverageEpisodeLength':
+                    tensor = tensor_util.make_ndarray(item.tensor)
+                    episode_lengths.append(int(tensor))
+
+    logging.info("Episode returns, %s, episode lengths: %s", episode_returns,
+                 episode_lengths)
+    return episode_returns, episode_lengths
+
+
+class TrainTest(parameterized.TestCase, unittest.TestCase):
+    def test_ac_cart_pole(self):
```

Comment:
It would be useful to have one test that verifies that training can run, e.g. set it to run only one or two iterations.